### PR TITLE
Fix tests recurrent

### PR DESCRIPTION
**Pull request by molbap**
# What does this PR do?

Continuation of #32649 since it seems RecurrentGemma tests failures were not caught

### CODE REVIEW

**Comment by molbap** [comment]
Only the tensorflow-examples are failing here, again, so let's see?

**Comment by amyeroberts** [comment]
@molbap Are there any previously passing tests which now are being skipped with this change?

**Comment by molbap** [comment]
I don't think there are - if `position_ids` were absent in the signature, this test would indeed have failed

**Comment by molbap** [comment]
Just checked, only additional tests skipped are those coming from RecurrentGemma as far as I can tell
In previous PR:
```bash
================ 2843 passed, 5777 skipped in 111.91s (0:01:51) ================
```
now:
```
=========== 1 failed, 2846 passed, 5773 skipped in 108.18s (0:01:48) ===========
```
That's on one out of 5 parallel runs, of course

**Review by amyeroberts — APPROVED** [review]
Thanks for fixing and checking the change in tests!

File: examples/tensorflow/summarization/run_summarization.py [hunk 1]
```diff
@@ -67,6 +67,14 @@
...
         )
     with FileLock(".lock") as lock:
+        try:
+            tokenizers_path = nltk.data.find("tokenizers").path
+        except (LookupError, OSError):
+            tokenizers_path = os.path.join(nltk.data.path[0], "tokenizers")
+            os.makedirs(tokenizers_path, exist_ok=True)
+        punkt_path = os.path.join(tokenizers_path, "punkt")
+        if os.path.exists(punkt_path):
+            os.remove(punkt_path)
```
Comment (amyeroberts) [inline examples/tensorflow/summarization/run_summarization.py:77]: waaaahhhh? we really shouldn't need to do this :/ 

Do we know why this is needed? e.g. examples interacting with each other / updated nltk

Comment (molbap) [reply examples/tensorflow/summarization/run_summarization.py:77]: Absolutely no idea why that happens, but let me try something else before merging this atrocity

Comment (molbap) [reply examples/tensorflow/summarization/run_summarization.py]: there we go, out with it - this test is weirdly flaky

Comment (molbap) [reply examples/tensorflow/summarization/run_summarization.py]: @amyeroberts removed this, think we can merge and see if this pops again on `main`

**Pull request merged**